# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.cpp
# Commit: c79511446521
# Full Hash: c795114465210fd2471ee91b68ba75c03afcbe40
# Author: stransky <stransky@redhat.com>
# Date: 2022-07-26 21:44:38
# Regressor Bug: 1777664
# File Overlap Count: 1
# Description:
#   Bug 1777664 [Wayland] Assert when ShowWaylandToplevelWindow() is called on popup r=emilio
#   
#   Depends on D152730
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152731
# ==============================================================================

diff -r 07d98fdfda1e -r c79511446521 widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp	Tue Jul 26 14:56:49 2022 +0000
+++ b/widget/gtk/nsWindow.cpp	Tue Jul 26 14:56:50 2022 +0000
@@ -1217,6 +1217,7 @@
 }
 
 void nsWindow::ShowWaylandToplevelWindow() {
+  MOZ_ASSERT(!IsWaylandPopup());
   LOG("nsWindow::ShowWaylandToplevelWindow");
   gtk_widget_show(mShell);
 }
